NFs/bsf/internal/sbi/processor/pcf_bindings.go

```diff
@@ -0,0 +1,350 @@
+		PcfId:              stringToPtr(request.PcfId),
+		PcfSetId:           stringToPtr(request.PcfSetId),
+		ParaCom:            request.ParaCom,
+		BindLevel:          (*models.BindingLevel)(&request.BindLevel),
+		Ipv4FrameRouteList: request.Ipv4FrameRouteList,
+		Ipv6FrameRouteList: request.Ipv6FrameRouteList,
+	}
+
+	if request.RecoveryTime != nil {
+		binding.RecoveryTime = request.RecoveryTime
+	}
+
+	// Check for existing binding based on combination
+	existingBindings := bsfContext.BsfSelf.QueryPcfBindings(
+		request.Supi,
+		request.Gpsi,
+		request.Dnn,
+		request.Ipv4Addr,
+		request.Ipv6Prefix,
+		request.MacAddr48,
+		request.IpDomain,
+		request.Snssai,
+	)
+
+	if len(existingBindings) > 0 {
+		// Return 403 with existing PCF binding info per 3GPP TS 29.521
+		problemDetail := models.ProblemDetails{
+			Status: http.StatusForbidden,
+			Cause:  "BINDING_ALREADY_EXISTS",
+			Detail: "Existing PCF binding information stored in BSF for the indicated combination",
+		}
+		c.JSON(http.StatusForbidden, problemDetail)
+		return
+	}
+
+	// Create new binding
+	bindingId := bsfContext.BsfSelf.CreatePcfBinding(binding)
+
+	// Convert back to response format
+	response := models.PcfBinding{
+		Supi:               ptrToString(binding.Supi),
+		Gpsi:               ptrToString(binding.Gpsi),
+		Ipv4Addr:           ptrToString(binding.Ipv4Addr),
+		Ipv6Prefix:         ptrToString(binding.Ipv6Prefix),
+		AddIpv6Prefixes:    binding.AddIpv6Prefixes,
+		IpDomain:           ptrToString(binding.IpDomain),
+		MacAddr48:          ptrToString(binding.MacAddr48),
+		AddMacAddrs:        binding.AddMacAddrs,
+		Dnn:                binding.Dnn,
+		PcfFqdn:            ptrToString(binding.PcfFqdn),
+		PcfIpEndPoints:     binding.PcfIpEndPoints,
+		PcfDiamHost:        ptrToString(binding.PcfDiamHost),
+		PcfDiamRealm:       ptrToString(binding.PcfDiamRealm),
+		PcfSmFqdn:          ptrToString(binding.PcfSmFqdn),
+		PcfSmIpEndPoints:   binding.PcfSmIpEndPoints,
+		Snssai:             binding.Snssai,
+		SuppFeat:           ptrToString(binding.SuppFeat),
+		PcfId:              ptrToString(binding.PcfId),
+		PcfSetId:           ptrToString(binding.PcfSetId),
+		ParaCom:            binding.ParaCom,
+		BindLevel:          (*binding.BindLevel),
+		Ipv4FrameRouteList: binding.Ipv4FrameRouteList,
+		Ipv6FrameRouteList: binding.Ipv6FrameRouteList,
+	}
+
+	if binding.RecoveryTime != nil {
+		response.RecoveryTime = binding.RecoveryTime
+	}
+
+	locationHeader := "/nbsf-management/v1/pcfBindings/" + bindingId
+	c.Header("Location", locationHeader)
+	c.JSON(http.StatusCreated, response)
+}
+
+// GetPCFBindings handles GET /pcfBindings
+func GetPCFBindings(c *gin.Context) {
+	logger.ProcLog.Infof("Handle GetPCFBindings")
+
+	// Extract query parameters
+	ipv4Addr := c.Query("ipv4Addr")
+	ipv6Prefix := c.Query("ipv6Prefix")
+	macAddr48 := c.Query("macAddr48")
+	dnn := c.Query("dnn")
+	supi := c.Query("supi")
+	gpsi := c.Query("gpsi")
+	ipDomain := c.Query("ipDomain")
+
```

Comment:
<img width="669" height="441" alt="image" src="https://github.com/user-attachments/assets/5cbc2810-8add-4fef-9ebb-db516c380a88" />

The `ipv4Addr`, `ipv6Prefix`, and `macAddr48` are conditional parameters. Please check those parameters as specified in NOTE 1.